Homework 4/Phone book/Phone book/Phone book.c

```diff
@@ -0,0 +1,236 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+
+typedef struct
+{
+    char name[20];
+    char phone[15];
+} PhoneBookEntry;
+
+void newEntry(int* indexForArray, PhoneBookEntry array[], const char* name, const char* phone)
+{
+    strcpy(array[*indexForArray].name, name);
+    strcpy(array[*indexForArray].phone, phone);
+    ++*indexForArray;
+}
+
+int findPhoneNumber(int numberOfEntries, PhoneBookEntry array[], char* name)
+{
+    int j = 0;
+    while (strcmp(array[j].name, name) != 0 && j < numberOfEntries)
```

Comment:
```suggestion
    while (j < numberOfEntries && strcmp(array[j].name, name) != 0)
```
Иначе мы сначала выполним чтение, а потом уже проверим, имеем ли мы на это право. Вычисление && и || выполняется слева направо